--- attempt4
+++ attempt5
@@ -118,7 +118,15 @@
       ∏ v ∈ s, coordMass A v (q v) := by
   calc
     _ = ∏ v, ∑ a, if v ∈ s → a = q v then coordMass A v a else 0 := by
-      convert cylinder_factorization A (fun v a => v ∈ s → a = q v) using 1 <;> rfl
+      convert cylinder_factorization A (fun v a => v ∈ s → a = q v) using 1
+      · apply Finset.sum_congr rfl
+        intro l _
+        split_ifs <;> rfl
+      · apply Finset.prod_congr rfl
+        intro v _
+        apply Finset.sum_congr rfl
+        intro a _
+        split_ifs <;> rfl
     _ = _ := ?_
   have hi (v : V) : (∑ a, if v ∈ s → a = q v then coordMass A v a else 0) =
       if v ∈ s then coordMass A v (q v) else 1 := by
